core/server/data/schema/schema.js

```diff
@@ -215,5 +215,21 @@ module.exports = {
         created_by: {type: 'integer', nullable: false},
         updated_at: {type: 'dateTime', nullable: true},
         updated_by: {type: 'integer', nullable: true}
+    },
+    invites: {
+        id: {type: 'increments', nullable: false, primary: true},
+        status: {type: 'string', maxlength: 150, nullable: false, defaultTo: 'pending', validations: {isIn: [['pending', 'sent']]}},
+        token: {type: 'string', maxlength: 191, nullable: false, unique: true},
+        email: {type: 'string', maxlength: 191, nullable: false, unique: true, validations: {isEmail: true}},
+        expires: {type: 'bigInteger', nullable: false},
+        created_at: {type: 'dateTime', nullable: false},
+        created_by: {type: 'integer', nullable: false},
+        updated_at: {type: 'dateTime', nullable: true},
+        updated_by: {type: 'integer', nullable: true}
+    },
+    invites_roles: {
+        id: {type: 'increments', nullable: false, primary: true},
+        role_id: {type: 'integer', nullable: false},
+        invite_id: {type: 'integer', nullable: false}
```

Comment:
My gut instinct here is that it seems a little over-the-top to have a whole table for storing the invite roles. Do you think it could work (and might actually be slightly simpler code) if this was a field in the invites table? 

Normally, a table would absolutely be the right approach, however in this case the roles are temporary data that needs to be stored until such time as the invite is accepted. Until converted to a user, the roles have no use. Therefore, it would seem to be enough to keep the role ID in a field on the `invites` table and to futureproof for the case of multiple roles, to have a comma separated list.